src/target/riscv/riscv.c

```diff
@@ -867,13 +877,32 @@ static int maybe_add_trigger_t2_t6(struct target *target,
 	}
 	LOG_TARGET_DEBUG(target, "trying to setup equality match trigger");
 	struct trigger_request_info eq = {
-		.tdata1 = fields.common |
-			(trigger->length == 1 ? fields.size.s8bit : fields.size.any) |
-			fields.chain.disable | fields.match.eq,
+		.tdata1 = fields.common | fields.size.any | fields.chain.disable |
+			fields.match.eq,
 		.tdata2 = trigger->address,
 		.tdata1_ignore_mask = fields.tdata1_ignore_mask
 	};
-	return try_setup_single_match_trigger(target, trigger, eq);
+	ret = try_setup_single_match_trigger(target, trigger, eq);
+	if (ret != ERROR_OK)
```

Comment:
@en-sc don't we need to handle ERROR_TARGET_RESOURCE_NOT_AVAILABLE return code here?

Reply:
Thanks for the catch! Addressed in #899